lms/djangoapps/mobile_api/testutils.py

```diff
@@ -123,8 +141,197 @@ def test_other_user(self):
+            field_data_cache,
+        )._xmodule
+        module.system.publish(self.problem_1, 'grade', grade_dict)
+
+    @patch.dict('django.conf.settings.FEATURES', {
+        'ENABLE_PREREQUISITE_COURSES': True,
+        'MILESTONES_APP': True,
+        'ENTRANCE_EXAMS': True
+    })
+    def test_feature_flags(self):
+        """
+        Tests when feature flags are set/unset, content is gated appropriately
+        """
+        self._add_prerequisite_course()
+        self._add_entrance_exam()
+        self.init_course_access()
+        if self.ALLOW_ACCESS_TO_MILESTONE_COURSE:
+            self.api_response(expected_response_code=None)
+        else:
+            response = self.api_response(expected_response_code=204)
+            self.assertEqual(response.data, self.MILESTONE_ERROR)
+        settings.FEATURES["MILESTONES_APP"] = False
+        self.api_response(expected_response_code=None)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_PREREQUISITE_COURSES': True, 'MILESTONES_APP': True})
+    def test_unfulfilled_prerequisite_course(self):
+        """ Tests the case for an unfulfilled pre-requisite course """
+        self._add_prerequisite_course()
+
+        self.init_course_access()
+        if self.ALLOW_ACCESS_TO_MILESTONE_COURSE:
+            self.api_response(expected_response_code=None)
+        else:
+            response = self.api_response(expected_response_code=204)
+            self.assertEqual(response.data, self.MILESTONE_ERROR)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_PREREQUISITE_COURSES': True, 'MILESTONES_APP': True})
+    def test_unfulfilled_prerequisite_course_for_staff(self):
+        self._add_prerequisite_course()
+
+        self.user.is_staff = True
+        self.user.save()
+        self.init_course_access()
+        self.api_response(expected_response_code=None)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_PREREQUISITE_COURSES': True, 'MILESTONES_APP': True})
+    def test_fulfilled_prerequisite_course(self):
+        """
+        Tests the case when a user fulfills existing pre-requisite course
+        """
+        self._add_prerequisite_course()
+
+        add_prerequisite_course(self.course.id, self.prereq_course.id)
+        fulfill_course_milestone(self.prereq_course.id, self.user)
+        self.init_course_access()
+        self.api_response(expected_response_code=None)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENTRANCE_EXAMS': True, 'MILESTONES_APP': True})
+    def test_unpassed_entrance_exam(self):
+        """
+        Tests the case where the user has not passed the entrance exam
+        """
+        self._add_entrance_exam()
+
+        self.init_course_access()
+
+        if self.ALLOW_ACCESS_TO_MILESTONE_COURSE:
+            self.api_response(expected_response_code=None)
+        else:
+            response = self.api_response(expected_response_code=204)
+            self.assertEqual(response.data, self.MILESTONE_ERROR)
```

Comment:
Line 306-310 are repeated several times in this Test class.  Please create a helper method for it.